certora/specs/GuardianStorage.spec

```
@@ -388,3 +389,14 @@ rule removeGuardianChangesGuardians {
     assert !isGuardian(safeContract, toRemove), "revokeGuardian should remove the given guardian";
     assert isGuardian(safeContract, other) == isGuardianOtherBefore, "revokeGuardian should not remove or add other guardians";
 }
+
+invariant harnessCountGuardiansCorrect(address wallet) 
+    ghostGuardianCount[wallet] == countGuardians(wallet)
```

```suggestion
    ghostGuardianCount[wallet] == countGuardians(wallet) &&
    guardiansCount(wallet) == countGuardians(wallet)
```

Should we have an additional check?

Run: https://prover.certora.com/output/950385/ecce437d39064827839ee0c024fc5d1a?anonymousKey=be069871dbd67ef64f4cf59e732d8c8ddbb5761e

---

Strange, for me it comes up with a counter example: https://prover.certora.com/output/6575/98c12605880742b8af941aca5a61ea8b?anonymousKey=dbc2fc37cce08cad1acf9ef606dd9aeed78d06ac

I think the pre-state is incorrect.

---

Ah, there were configuration changes related to loops, I brought optimistic_loop option back and now it works. Thanks for the suggestion!